p2p/exchange.go

```diff
@@ -155,12 +156,20 @@ func (ex *Exchange[H]) Head(ctx context.Context, opts ...header.HeadOption[H]) (
 			if useTrackedPeers {
 				err = reqParams.TrustedHead.Verify(headers[0])
 				if err != nil {
-					log.Errorw("verifying head received from tracked peer", "tracked peer", from,
-						"height", headers[0].Height(), "err", err)
+					var verErr *header.VerifyError
+					if errors.As(err, &verErr) && verErr.SoftFailure {
```

Comment:
We should also be calling Verify defined in the root

Reply:
https://github.com/celestiaorg/go-header/pull/109/commits/9b2f384fc180be4ee164ec56a1ca9fdacc06cdef

Reply:
This might now "break" bc i changed the param type

Reply:
It is fine Header[H] == H